gensim/utils.py

```diff
@@ -1263,3 +1269,18 @@ def _iter_windows(document, window_size, copy=False, ignore_below_size=True):
     else:
         for doc_window in doc_windows:
             yield doc_window.copy() if copy else doc_window
+
+
+def walk_with_depth(top, topdown=True, onerror=None, followlinks=False):
+    """Wrap `os.walk` in code that analyzes the directory path of each yield to
+    determine the depth relative to `top`. Yields in the `top` directory are at
+    depth 0.
+
+    Returns:
+        generator of tuples of (depth, dirpath, dirnames, filenames).
+    """
+    path = os.path.abspath(top)
+    for dirpath, dirnames, filenames in os.walk(path, topdown, onerror, followlinks):
+        sub_path = dirpath.replace(path, '')
+        depth = sub_path.count(os.sep)
```

Comment:
Ah, I see your concern. I think `os.path.abspath` handles the situation you're worried about:
```
In [4]: os.path.abspath('/test/path/')
Out[4]: '/test/path'

In [5]: os.path.abspath('/test/path')
Out[5]: '/test/path'

In [6]: os.path.abspath('/test/path//')
Out[6]: '/test/path'
```